### ISSUE TITLE
NumberFormatException : Payment Gateway `order` property

### ISSUE THREAD

**Issue by AmandaRiu** [woocommerce/woocommerce-android#1840]
The `GatewayMapper` in FluxC expects `response.order` to contain an `Int` but it may also contain an empty string which is strange, but really happening. This is causing the following error:

```
[Jan-13 10:42 WP e] API Volley error on https://public-api.wordpress.com/rest/v1.1/jetpack-blogs/85725384/rest-api/?path=%2Fwc%2Fv3%2Fpayment_gateways%2F%26_method%3Dget&json=true&locale=nl_NL - exception: com.google.gson.JsonSyntaxException: java.lang.NumberFormatException: empty String
[Jan-13 10:42 WP e] API StackTrace: com.android.volley.ParseError: com.google.gson.JsonSyntaxException: java.lang.NumberFormatException: empty String
	at org.wordpress.android.fluxc.network.rest.GsonRequest.parseNetworkResponse(GsonRequest.java:98)
	at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:143)
	at com.android.volley.NetworkDispatcher.processRequest(NetworkDispatcher.java:111)
	at com.android.volley.NetworkDispatcher.run(NetworkDispatcher.java:90)
Caused by: com.google.gson.JsonSyntaxException: java.lang.NumberFormatException: empty String
	at com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:228)
	at com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:218)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
	at com.google.gson.internal.bind.TypeAdapterRuntimeTypeWrapper.read(TypeAdapterRuntimeTypeWrapper.java:41)
	at com.google.gson.internal.bind.ArrayTypeAdapter.read(ArrayTypeAdapter.java:72)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$1.read(ReflectiveTypeAdapterFactory.java:131)
	at com.google.gson.internal.bind.ReflectiveTypeAdapterFactory$Adapter.read(ReflectiveTypeAdapterFactory.java:222)
	at com.google.gson.Gson.fromJson(Gson.java:927)
	at com.google.gson.Gson.fromJson(Gson.java:892)
	at com.google.gson.Gson.fromJson(Gson.java:841)
	at org.wordpress.android.fluxc.network.rest.GsonRequest.parseNetworkResponse(GsonRequest.java:90)
	... 3 more
Caused by: java.lang.NumberFormatException: empty String
	at sun.misc.FloatingDecimal.readJavaFormatString(FloatingDecimal.java:1842)
	at sun.misc.FloatingDecimal.parseDouble(FloatingDecimal.java:110)
	at java.lang.Double.parseDouble(Double.java:538)
	at com.google.gson.stream.JsonReader.nextInt(JsonReader.java:1201)
	at com.google.gson.internal.bind.TypeAdapters$7.read(TypeAdapters.java:226)
	... 14 more
```

Example json:
```
{
         "id":"mollie_wc_gateway_sofort",
         "title":"SOFORT Banking",
         "description":"",
         "order":12,
         "enabled":true, ...
},
{
         "id":"mollie_wc_gateway_directdebit",
         "title":"SEPA-incasso",
         "description":"",
         "order":"",
         "enabled":true, ...
}
```

see 2627677-zen for additional background.

**Comment by AmandaRiu** [comment]
Note: be sure to catch empty strings as well as nulls before converting to Int otherwise it will crash:
```
response.order.takeUnless { it.isNullOrEmpty() }?.toInt() ?: 0,
```

**Comment by nbradbury** [comment]
Fixed via #1847
